transport/grpc/outbound_test.go

```diff
@@ -200,3 +203,104 @@ func TestCallStreamWithInvalidPeer(t *testing.T) {
 		}.Error(),
 	)
 }
+
+func TestCallSvcMatch(t *testing.T) {
+	const (
+		successType  = "success"
+		failType     = "fail"
+		noheaderType = "noheader"
+	)
+	tests := []struct {
+		msg        string
+		serverPort string
+		testType   string
+		wantErr    bool
+	}{
+		{
+			"test call service match success",
+			"15213",
+			successType,
+			false,
+		},
+		{
+			"test call service match failed",
+			"15619",
+			failType,
+			true,
+		},
+		{
+			"test no service name response header",
+			"15640",
+			noheaderType,
+			false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.msg, func(t *testing.T) {
+			server := grpc.NewServer(
+				grpc.CustomCodec(customCodec{}),
+				grpc.UnknownServiceHandler(func(srv interface{}, stream grpc.ServerStream) error {
+					ctx := stream.Context()
+					// Get transport request
+					md, ok := metadata.FromIncomingContext(ctx)
+					if md == nil || !ok {
+						return yarpcerrors.Newf(yarpcerrors.CodeInternal, "cannot get metadata from ctx: %v", ctx)
+					}
+					transportRequest, err := metadataToTransportRequest(md)
+					if err != nil {
+						return err
+					}
+
+					responseWriter := newResponseWriter()
+					defer responseWriter.Close()
+
+					if tt.testType == successType {
```

Comment:
Actually, let's simplify this a bit more:
Instead of using the `const`s we defined at the top, let's add a two `string` fields to our test struct a `key` and `value`.

This test will then always write an arbitrary system header using those two values. For the no header case, we can just use empty strings.